### ISSUE TITLE
the repairing process is still not completely done and manages only to initialize but not to connect

### ISSUE THREAD

**Issue by GameBully2K** [rajatK012/hashconnectWalletConnect#6]
the local data is readed correctly but when it tries to connect with it errors
![image](https://user-images.githubusercontent.com/71235565/170719030-0ae59c4c-3381-43c7-a63d-ef72b04b0eef.png)

**Comment by Visuals3D** [comment]
Yes its just missing a await here to wait for the init to finish: 
```
  hashConnect.init(metaData ?? APP_CONFIG, localData?.privateKey);
  hashConnect.connect(
    localData?.topic,
    localData?.pairedWalletData ?? metaData
  );
```
change it to:
```
  await hashConnect.init(metaData ?? APP_CONFIG, localData?.privateKey);
  hashConnect.connect(
    localData?.topic,
    localData?.pairedWalletData ?? metaData
  );
```

**Comment by edufolios-matt** [comment]
Thanks @Visuals3D that helped me with some similar issues I was seeing this side too. I'm still seeing that the pairedAccounts, pairingString and privateKey are blank / null in the walletData of the object after the initial connect & reload though. Any thoughts?

**Comment by woodwardmatt** [comment]
Thanks @Visuals3D that helped me with some similar issues I was seeing this side too. I'm still seeing that the pairedAccounts, pairingString and privateKey are blank / null in the walletData of the object after the initial connect & reload though. Any thoughts?

**Comment by Visuals3D** [comment]
@woodwardmatt We encountered multiple other problems with this example so we didn't investigated further on this sorry.
The angular example from hashconnect is a good startingpoint to build your own service or provider in react: 
https://github.com/Hashpack/hashconnect/tree/main/example/dapp
